OOP RUBY - User interface

teekaytech opened the pull request:
### In this milestone, I did the following

- Created a new branch called user_interface
- Added contents to main.rb file for inputs and outputs
- Mimic the basic flow of tic tac toe game
- Have a deeper understanding of the game flow

## Kindly review for approval.

shshamim63 commented on bin/main.rb:
```diff
@@ -1,2 +1,54 @@
+puts "\nWith each boxes displayed from 1 to 9, each player can type in a value >=1 and <= 9"
+
+count = 0
+while count < 9
```
[Suggestion for your next milestone]
Your loop should not depend on any constant number. Cause it reduces the scalability of your code.

teekaytech replied:
This will be taken care of in the next milestone.

shshamim63 commented on bin/main.rb:
```diff
@@ -1,2 +1,54 @@
+
+count = 0
+while count < 9
+  if count.even?
+    loop do
+      print "#{player_one}, select your move: "
+      input = gets.chomp.to_i
+      # the check below ensures that user input is within 1 - 9 to be displayed on the board
+      if (input >= 1) && (input <= 9)
+        p1_move << input
+        # After a move, a check is run to see if the move have selected. If not, the move is displayed on the board
+        break
+      else
+        print 'Invalid input, try again! '
+      end
+    end
+  else
+    loop do
+      print "#{player_two}, select your move: "
+      input = gets.chomp.to_i
+      if (input >= 1) && (input <= 9)
+        p2_move << input
+        break
+      else
+        print 'Invalid input, try again! '
+      end
+    end
+  end
```
[Optional]
both if and else contains the same code except the variables. It is the right place to introduce a method to cover this task.

teekaytech replied:
I already have a plan to change this when I start the game logic in milestone 3.

teekaytech commented:
### Thank you, @shshamim63 for our response.

I have added the required scenario from your previous review.

Kindly review again, for approval.

kobiyoyo commented on bin/main.rb:
```diff
@@ -1,2 +1,60 @@
+puts "\nWith each boxes displayed from 1 to 9, each player can type in a value >=1 and <= 9"
+
+count = 0
+while count < 9
```
As previous TSE specified you need to use a variable to terminate the loop not a number.for example
````ruby

game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
`````

teekaytech replied:
Fixed.

teekaytech commented:
### Thank you, @kobiyoyo  for your response.
I have fixed all the required issues raised. I did the following:
- I ensured that the game informs the players if the selected move is invalid
- I also ensured that the game displays board after the player move
- I added the feature for the game to inform the players if the selected move is a winning move
- In addition, I ensured that the game informs the player if it ended in a draw.
- Lastly, I used a variable to terminate the loop, not a number.

Kindly review again, for approval. :wavy_dash:
